tools/docker/builder/rdkb/toolchain/build.sh

```diff
@@ -11,23 +11,51 @@ rootdir="${scriptdir%/*/*/*/*/*}"
 
 . ${rootdir}/tools/docker/functions.sh
 
+usage() {
+    echo "usage: $(basename $0) [-hrv]"
+    echo "  options:"
+    echo "      -h|--help - show this help menu"
+    echo "      -r|--rdkb-root - path to rdkb root (default is PWD)"
+    echo "      -v|--verbose - verbosity on"
+    echo ""
+}
+
+build_image() {
+    # We first need to build the corresponding images
+    docker build --tag rdkb-builder-${SUDO_USER:-${USER}} \
+           --build-arg USER_ID=${SUDO_UID:-$(id -u)} \
+           --build-arg GROUP_ID=${SUDO_GID:-$(id -g)} \
+           "$scriptdir/"
+}
+
 main() {
 
-    if [ -z "$1" ]; then
-        err "Usage: $0 <rdkb_dir> <command>"
-        exit 1
-    fi   
-    RDKB_ROOT=$(realpath $1)
-    shift
+    OPTS=`getopt -o 'hir:v' --long help,image,rdkb-path:,verbose -n 'parse-options' -- "$@"`
 
+    if [ $? != 0 ] ; then err "Failed parsing options." >&2 ; usage; exit 1 ; fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -h | --help)               usage; exit 0; shift ;;
+            -r | --rdkb-root)          RDKB_ROOT="$2"; shift; shift ;;
+            -v | --verbose)            VERBOSE=true; shift ;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1 ;;
+        esac
+    done
+
+    build_image
     # Default docker arguments
     docker_args="\
-     --workdir=$PWD --user=${SUDO_UID:-$(id -u)}:${SUDO_GID:-$(id -g)} \
-     -e USER=${SUDO_USER:-${USER}} -e HOME=${HOME} -v ${HOME}:${HOME} \
```

Comment:
Did the script really used to mount the whole "$HOME" inside the container? Why not just the folder you need?

Reply:
Fixed to use a single image.